src/service/userService.ts

```diff
@@ -0,0 +1,126 @@
+import UserModel from "../model/userModel";
+import loggerManager from "../utility/logger";
+import errorEnums from "../constants/errorConstants";
+import successEnums from "../constants/successConstant";
+import { genSaltSync, hashSync, compareSync } from "bcryptjs";
+import bcrypt from "bcrypt";
+import jwt from "jsonwebtoken";
+
+const logger = loggerManager.getLogger();
+
+interface UserInterface {
+  id?: number;
+  name: string;
+  email: string;
+  password: string;
+  role: string;
+}
+
+class UserService {
+  public async getAllUsers() {
+    try {
+      const allUsers: UserInterface[] = await UserModel.findAll({ raw: true });
+      return allUsers;
+    } catch (error) {
+      logger.error(error.message);
+      logger.error(errorEnums.INT_SERVER_ERR);
+    }
+  }
+
+  public async getUserById(id: number) {
+    const user = await UserModel.findByPk(id);
+    if (!user) {
+      logger.error(errorEnums.ERR_INVALID_INPUT);
+    }
+    return user;
+  }
+
+  public async createNewUser(
+    name: string,
+    email: string,
+    password: string,
+    role: string
+  ) {
+    try {
+      const salt = genSaltSync(10);
+      const hashPassword = hashSync(password, salt);
+      const newUser: UserInterface = await UserModel.create({
+        name: String(name),
+        email: String(email),
+        password: String(hashPassword),
+        role: String(role),
+      });
+
+      return newUser;
+    } catch (error) {
+      logger.error(errorEnums.INT_SERVER_ERR);
+    }
+  }
+
+  public async updateUserById(
+    id: number,
+    name: string,
+    email: string,
+    password: string
+  ) {
+    try {
+      const salt = genSaltSync(10);
+      const hashPassword = hashSync(password, salt);
+      const [affectedRows] = await UserModel.update(
+        {
+          name: String(name),
+          email: String(email),
+          password: String(hashPassword),
+        },
+        { where: { id } }

```

Comment:
there should separate method to update each of these  attributes of the User model as these are sensitive information of user